### DISCUSSION TITLE
Passing `trainer` engine to tb_logger attach of `evaluator` engine.

### DISCUSSION THREAD

**Discussion by aksg87** [pytorch/ignite#2031] [Q&A] [answered]
While trying Ignite (I love it!) I came across a situation where I wanted to pass the trainer engine into a call to `tb_logger.attach` on an evaluator engine. I did this using `functools.partial` but I wanted to confirm there isn't a better way.

Example:

```
from functools import partial

...

# some custom handler. 
def predictions_gt_images_handler(engine, logger, *args, **kwargs):

    x = engine.state.batch["image"]

    y_pred, y = engine.state.output

    # ... more handler code ...

    trainer = kwargs["trainer"]  # passed in as **kwargs
    logger.writer.add_figure(
        "predictions vs actuals", figure=fig, global_step=trainer.state.epoch  # I would like to use another engine
    )


with TensorboardLogger(log_dir="quick-start-segmentation") as tb_logger:
    predictions_gt_images_handler_bound = partial(
        predictions_gt_images_handler, trainer=trainer
    )  # binds trainer to **kwargs -- *** Is there a cleaner approach *** ?

    tb_logger.attach(
        evaluator,
        log_handler=predictions_gt_images_handler_bound,
        event_name=Events.EPOCH_COMPLETED,
    )

**Comment by vfdev-5** [comment, marked as answer]
@aksg87 thanks for the question and the feedback about the library ! 

Unfortunately, in the current state of the library, we can not propagate args/kwargs to the handler but I could be easy to add as a new feature.

Currently, instead of using `tb_logger.attach` you can write it like that:
```python
with TensorboardLogger(log_dir="quick-start-segmentation") as tb_logger:

    e = Events.EPOCH_COMPLETED 
    evaluator.add_event_handler(
        e, predictions_gt_images_handler, tb_logger, e, trainer=trainer
    )
```


PS: by the way, in case if you haven't seen previously our message about a survey:
> we would love to hear user's feedback on our documentation! The survey will only take around 1-2 minutes to complete but your input will go a long way in improving our docs.
Take the survey: https://cutt.ly/ignite-doc-survey

**Comment by aksg87** [comment]
Thanks @vfdev-5  -- I completed the survey as well :)
